### PR TITLE
feat(wallet): remove Goerli network

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/37369
Resolves https://github.com/brave/brave-browser/issues/34920
Supersedes https://github.com/brave/brave-core/pull/21403

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Before testing, ensure that you have launched an older version of Brave with Goerli set as the default EVM network.
2. Launch Brave with changes from this PR
3. The current network should be set to Sepolia. Goerli network should not be visible in the portfolio.

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by iefremov — APPROVED** [review]
unlocking renderer/*

**Review by nuo-xu — APPROVED** [review]
iOS++

File: components/brave_wallet_ui/common/slices/api-slice-extras-tests/use-combined-tokens-list.test.ts [hunk 1]
```diff
@@ -47,6 +47,6 @@ describe('useCombinedTokensList', () => {
 
     // done loading
     expect(result.current.isLoading).toBe(false)
-    expect(result.current.data).toHaveLength(19)
+    expect(result.current.data).toHaveLength(18)
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/slices/api-slice-extras-tests/use-combined-tokens-list.test.ts:50]: which coin was removed?

Comment (onyb) [reply components/brave_wallet_ui/common/slices/api-slice-extras-tests/use-combined-tokens-list.test.ts:50]: The native asset of Goerli.

**Review by josheleonard — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java [hunk 2]
```diff
@@ -131,7 +131,7 @@ public static void getAllTokens(
         blockchainRegistry.getAllTokens(
                 chainId,
                 coinType,
-                tokens -> callback.call(Utils.fixupTokensRegistry(tokens, chainId)));
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java:134]: The method `TokenUtils.getAllTokens` is now useless as it immediately calls `BlockchainRegistry.getAllTokens`.
We should completely remove this method and replace its occurrences with  `blockchainRegistry.getAllTokens`.

Comment (onyb) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java:134]: Fixed here: https://github.com/brave/brave-core/pull/24438/commits/cd9e87f7eb2a50bdd4370a7c131a39ccbfcfa134

**Review by simoarpe — APPROVED** [review]
👍 Android changes look good!

**Review by supermassive — APPROVED** [review]
wallet core lgrm

**Comment by srirambv** [comment]
Verification passed on
 
 Brave | 1.70.20 Chromium: 127.0.6533.73 (Official Build) nightly (64-bit)
 -- | --
 Revision | 63a9d9566501645c5defb083cba496054b532d86
 OS | Linux
 
 - Verified steps from PR description
 - Verified setting up Goerli as default on older version and upgrading to 1.70.20 removes Goerli from Portfolio and sets Sepolia as default
 - Verified manually adding Goerli with Ankr backend enabled shows balance
 
 https://github.com/user-attachments/assets/f44bc4ff-f23e-4513-b129-e6cbff3ab27a

**Pull request merged**